File: templates/cpp/bot.cpp
// Your Connect Four bot. This is the only file you need to edit (along with
// the function declarations in bot.hpp, if you add helpers of your own).
//
// The HTTP server lives in server.cpp. It calls `choose_move` once per turn
// with the current board and sends your answer back to the arena.
#include "bot.hpp"

#include <random>

int choose_move(const std::vector<std::vector<int>>& board, int you) {
  (void)you;  // the default bot plays randomly and doesn't care who it is

  std::vector<int> moves = legal_moves(board);

  static std::mt19937 rng(std::random_device{}());
  std::uniform_int_distribution<size_t> dist(0, moves.size() - 1);
  return moves[dist(rng)];
}

std::vector<int> legal_moves(const std::vector<std::vector<int>>& board) {
  // Columns that aren't full yet (top row is still empty).
  std::vector<int> moves;
  for (size_t col = 0; col < board.size(); ++col) {
    if (board[col].back() == 0) {
      moves.push_back(static_cast<int>(col));
    }
  }
  return moves;
}

File: templates/go/bot.go
// Your Connect Four bot. This is the only file you need to edit.
//
// The HTTP server lives in server.go. It calls chooseMove once per turn
// with the current board and sends your answer back to the arena.
package main

import "math/rand"

// chooseMove picks your next move.
//
// board is a slice of 8 columns, each a slice of 8 rows: board[col][row].
// Column 0 is the LEFT column, row 0 is the BOTTOM row.
// 0 = empty, 1 = player 1's piece, 2 = player 2's piece.
//
// you is 1 or 2: which player you are this game.
//
// Return an int 0-7, the column you want to drop a piece into. It MUST be a
// legal (non-full) column; see legalMoves below.
//
// Every call gets the full game state, so don't rely on variables that
// persist between calls: the arena may restart your bot mid-game.
func chooseMove(board [][]int, you int) int {
	moves := legalMoves(board)
	return moves[rand.Intn(len(moves))]
}

// legalMoves returns the columns that aren't full yet (top row is still empty).
func legalMoves(board [][]int) []int {
	var moves []int
	for col, column := range board {
		if column[len(column)-1] == 0 {
			moves = append(moves, col)
		}
	}
	return moves
}
